### DISCUSSION TITLE
v3-b27: $table(..., layout: ...) - layout expects string, documentation shows enum

### DISCUSSION THREAD

**Discussion by wivaku** [filamentphp/filament#7338] [Help]
code: `HasFilters.php` shows `filters` accepts `layout` argument (string)
```
    public function filters(array $filters, string | Closure | null $layout = null): static
```

documentation shows it using new enum: 
https://beta.filamentphp.com/docs/3.x/tables/filters#displaying-filters-above-the-table-content
```
use Filament\Tables\Enums\FiltersLayout;
use Filament\Tables\Table;
 
public function table(Table $table): Table
{
    return $table
        ->filters([
            // ...
        ], layout: FiltersLayout::AboveContent);
}
```

Related, no documentation for the method (in `HasFilters.php`) that does accept the enum:
(except for the [global table settings](https://beta.filamentphp.com/docs/3.x/tables/advanced#global-settings))
```
    public function filtersLayout(FiltersLayout | Closure | null $filtersLayout): static
    {
        $this->filtersLayout = $filtersLayout;

        return $this;
    }
```

Related, worth mentioning in the upgrade documentation that this is a breaking change.
Argument `filters` used to take `Filament\Tables\Filters\Layout` instead of the new `Filament\Tables\Enums\FiltersLayout`.

**Comment by danharrin** [comment]
fixing it so it accepts an enum
